File: 01-10-2025_Bottles.py
class Solution:
    def numWaterBottles(self, n: int, x: int) -> int:
        total = 0
        empty_bottles = 0
        while n > 0:
            total += n
            empty_bottles += n
            n = empty_bottles // x
            empty_bottles = empty_bottles % x
        return total

File: 01-10-2025_Bottles.cpp
class Solution {
public:
    int numWaterBottles(int n, int x) {
        int sum=0;
        int eb=0;
        while(n)
        {
            sum+=n;
            eb+=n;
             n=eb/x;
            eb=eb%x;
        }
        return sum;
    }
};